In [99]:
import pandas as pd

pd.set_option("display.max_columns", None)


# https://www.kaggle.com/datasets/bartoszpieniak/poland-cars-for-sale-dataset
FILE_PATH = "../scripts/data/Car_sale_ads.csv"

In [100]:
df = pd.read_csv(FILE_PATH)
df.head(20)

Index   Price Currency Condition Vehicle_brand Vehicle_model Vehicle_version Vehicle_generation  Production_year  Mileage_km  Power_HP  Displacement_cm3 Fuel_type  CO2_emissions         Drive Transmission        Type  Doors_number  Colour Origin_country First_owner First_registration_date Offer_publication_date                                     Offer_location                                           Features
0       0   86200      PLN       New        Abarth           595             NaN                NaN             2021         1.0     145.0            1400.0  Gasoline            NaN  Front wheels       Manual  small_cars           3.0    gray            NaN         NaN                     NaN             04/05/2021  ul. Jubilerska 6 - 04-190 Warszawa, Mazowiecki...                                                 []
1       1   43500      PLN      Used        Abarth         Other             NaN                NaN             1974     59000.0      75.0            1100.0  Gasoline            NaN  Front wheels       Manual       coupe           2.0  silver            NaN         NaN                     NaN             03/05/2021  kanonierska12 - 04-425 Warszawa, Rembertów (Po...                                                 []
2       2   44900      PLN      Used        Abarth           500             NaN                NaN             2018     52000.0     180.0            1368.0  Gasoline            NaN           NaN    Automatic  small_cars           3.0  silver            NaN         NaN                     NaN             03/05/2021                   Warszawa, Mazowieckie, Białołęka  ['ABS', 'Electric front windows', 'Drivers air...
3       3   39900      PLN      Used        Abarth           500             NaN                NaN             2012     29000.0     160.0            1368.0  Gasoline          139.0  Front wheels       Manual  small_cars           3.0    gray            NaN         NaN                     NaN             30/04/2021                                  Jaworzno, Śląskie  ['ABS', 'Electric front windows', 'Drivers air...
4       4   97900      PLN       New        Abarth           595             NaN                NaN             2021       600.0     165.0            1368.0  Gasoline            NaN           NaN       Manual  small_cars           3.0    blue            NaN         NaN                     NaN             30/04/2021  ul. Gorzysława 9 - 61-057 Poznań, Nowe Miasto ...  ['ABS', 'Electrically adjustable mirrors', 'Pa...
5       5   62950      PLN      Used        Abarth           595             NaN                NaN             2016     46060.0     180.0            1368.0  Gasoline          139.0  Front wheels       Manual  small_cars           3.0    gray        Germany         Yes              15/03/2016             30/04/2021  Modlińska 157 - 03-186 Warszawa, Białołęka (Po...  ['ABS', 'Electric front windows', 'Drivers air...
6       6   69900      PLN      Used        Abarth           595             NaN                NaN             2021      2900.0     145.0            1368.0  Gasoline            NaN           NaN       Manual  small_cars           2.0   black        Germany         NaN              25/02/2021             29/04/2021                             Żary, żarski, Lubuskie                                                 []
7       7   69900      PLN      Used        Abarth           500             NaN                NaN             2019     20000.0     145.0            1368.0  Gasoline          139.0  Front wheels       Manual  small_cars           3.0   white         Poland         Yes                     NaN             29/04/2021     Ojcowska 2 - 02-918 Warszawa, Mokotów (Polska)  ['ABS', 'Electrically adjustable mirrors', 'Pa...
8       8  120750      PLN       New        Abarth           595             NaN                NaN             2021         1.0     180.0            1368.0  Gasoline            NaN  Front wheels       Manual  small_cars           3.

In [101]:
# Print column information
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208304 entries, 0 to 208303
Data columns (total 25 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    208304 non-null  int64  
 1   Price                    208304 non-null  int64  
 2   Currency                 208304 non-null  object 
 3   Condition                208304 non-null  object 
 4   Vehicle_brand            208304 non-null  object 
 5   Vehicle_model            208304 non-null  object 
 6   Vehicle_version          138082 non-null  object 
 7   Vehicle_generation       147860 non-null  object 
 8   Production_year          208304 non-null  int64  
 9   Mileage_km               207321 non-null  float64
 10  Power_HP                 207661 non-null  float64
 11  Displacement_cm3         206338 non-null  float64
 12  Fuel_type                208304 non-null  object 
 13  CO2_emissions            94047 non-null   float64
 14  Driv

Index         Price  Production_year    Mileage_km       Power_HP  Displacement_cm3  CO2_emissions   Doors_number
count  208304.000000  2.083040e+05    208304.000000  2.073210e+05  207661.000000     206338.000000   9.404700e+04  206817.000000
mean   104151.500000  6.305383e+04      2012.098241  1.502768e+05     151.836281       1882.567147   3.191564e+05       4.637138
std     60132.329574  8.665967e+04         6.998414  2.937447e+06      77.683546        729.609660   7.291396e+07       0.768559
min         0.000000  5.000000e+02      1915.000000  1.000000e+00       1.000000        400.000000   1.000000e+00       1.000000
25%     52075.750000  1.780000e+04      2008.000000  5.300000e+04     105.000000       1461.000000   1.200000e+02       5.000000
50%    104151.500000  3.570000e+04      2013.000000  1.445660e+05     136.000000       1798.000000   1.400000e+02       5.000000
75%    156227.250000  7.599000e+04      2017.000000  2.060000e+05     172.000000       1997.000000   1.640000e+02       5.000000
max    208303.000000  6.999000e+06      2021.000000  1.111111e+09    1398.000000       8400.000000   2.000000e+10      55.000000

# Display data characteristics to determine column processing


In [102]:
# See the percentage of drive types for given car model
df["Vehicle"] = df["Vehicle_brand"] + "/" + df["Vehicle_model"]

# Group by the new Vehicle column and Drive, then calculate the size of each group
drive_counts = df.groupby(["Vehicle", "Drive"]).size().reset_index(name="count")

# Calculate the total count for each Vehicle
total_counts = df.groupby("Vehicle")["Drive"].count().reset_index(name="total_count")

# Merge the counts with the total counts
drive_counts = drive_counts.merge(total_counts, on="Vehicle")

# Calculate the percentage
drive_counts["percentage"] = (drive_counts["count"] / drive_counts["total_count"]) * 100

# Display the result
drive_counts

Vehicle                         Drive  count  total_count  percentage
0              Abarth/124                   Rear wheels      1            1  100.000000
1              Abarth/500                  Front wheels      9            9  100.000000
2              Abarth/595                  Front wheels     31           31  100.000000
3     Abarth/Grande Punto                  Front wheels      5            5  100.000000
4            Abarth/Other                  Front wheels      4            4  100.000000
...                   ...                           ...    ...          ...         ...
2264         Škoda/Superb                   Rear wheels      1         1823    0.054855
2265           Škoda/Yeti  4x4 (attached automatically)     19          160   11.875000
2266           Škoda/Yeti               4x4 (permanent)     34          160   21.250000
2267           Škoda/Yeti                  Front wheels    107          160   66.875000
2268            Żuk/Other                   Rear wheels      2            2  100.000000

[2269 rows x 5 columns]

In [103]:
# See the percentage of transmission types for given car model
df["Vehicle"] = df["Vehicle_brand"] + "/" + df["Vehicle_model"]

# Group by the new Vehicle column, Year, and Transmission, then calculate the size of each group
transmission_counts = (
    df.groupby(["Vehicle", "Production_year", "Transmission"]).size().reset_index(name="count")
)

# Calculate the total count for each Vehicle and Year
total_counts = (
    df.groupby(["Vehicle", "Production_year"])["Transmission"]
    .count()
    .reset_index(name="total_count")
)

# Merge the counts with the total counts
transmission_counts = transmission_counts.merge(total_counts, on=["Vehicle", "Production_year"])

# Calculate the percentage
transmission_counts["percentage"] = (
    transmission_counts["count"] / transmission_counts["total_count"]
) * 100

# Display the result
transmission_counts

Vehicle  Production_year Transmission  count  total_count  percentage
0      Abarth/124             2016       Manual      1            1       100.0
1      Abarth/124             2018       Manual      1            1       100.0
2      Abarth/500             2008       Manual      1            1       100.0
3      Abarth/500             2011    Automatic      1            1       100.0
4      Abarth/500             2012       Manual      2            2       100.0
...           ...              ...          ...    ...          ...         ...
13501   Żuk/Other             1970       Manual      2            2       100.0
13502   Żuk/Other             1977       Manual      2            2       100.0
13503   Żuk/Other             1985       Manual      2            2       100.0
13504   Żuk/Other             1986       Manual      2            2       100.0
13505   Żuk/Other             2014       Manual      2            2       100.0

[13506 rows x 6 columns]

## Field handling

### **Vehicle** (205,160 / 208,304 non-null)

**Description:** Contains "brand/model" information (e.g. Alfa RomeoGiulietta **1.4 TB 16V**).

**Importance:** Field is not important. It duplicates code from otherfields.

**Handling N/A:** REMOVE COLUMN


### **Vehicle_version** (138,082 / 208,304 non-null)

**Description:** Contains information about model version (e.g. Alfa RomeoGiulietta **1.4 TB 16V**).

**Importance:** Field contains important values, due to value difference for specific trims.

**Handling N/A:** MERGE vehicle model and version as a extended column (e.g. advanced_version)

### **Vehicle_generation** (147,860 / 208,304 non-null)
**Description:** Contains information about model generation.

**Importance:** Field might be sometimes important, because some generations might be more desired than others.

**Handling N/A:** FILL EMPTY WITH UKNOWN


### **Mileage_km** (207,321 / 208,304 non-null)
**Description:** Contains information about car's mileage.

**Importance:** Field is very important, due to high influence of mileage on cars value.

**Handling N/A:** REMOVE ROWS: As the car's without given mileage are usually not representative for average case, they are outliers that do not provide meaningful information.


### **Power_HP** (207,661 / 208,304 non-null)

**Description:** Contains information about car's horsepower.

**Importance:** Field is very important, due to high influence of engine power on price (more power == better car trim)

**Handling N/A:** DELETE ROWS - As there are not many of them and displacement is usually missing for them too, it's better to remove these rows.


### **Displacement_cm3** (206,338 / 208,304 non-null)

**Description:** Contains information about car's engine displacement in cm3

**Importance:** Field is rather very important, due to high influence of displacement on price (more displacement == more power == better car trim)

**Handling N/A:** DELETE ROWS - As there are not many of them and displacement is usually missing for them too, it's better to remove these rows.


### **CO2_emissions** (94,047 / 208,304 non-null)

**Description:** Contains information about car's engine displacement in cm3

**Importance:** Field is not important for car value.

**Handling N/A:** DELETE COLUMN - not important for evaluation


### **Drive** (94,047 / 208,304 non-null)

**Description:** Contains information about car's drive (eg. 4-wd, front-wheel drive, rear-wheel drive)

**Importance:** Field is important as it usually means a rather significant price difference

**Handling N/A:** FILL WITH MOST FREQUENT VALUE - from analyzing data above, most car models tend to favor certain drive types.

### **Transmission** (94,047 / 208,304 non-null)

**Description:** Contains information about car's transmission (e.g. manual, automatic)

**Importance:** Similar to drive

**Handling N/A:** FILL WITH MOST FREQUENT VALUE - similar to drive

### **Doors_number** (206,817 / 208,304 non-null)

**Description:** Contains information about car's door count

**Importance:** Not very important

**Handling N/A:** FILL WITH MOST FREQUENT VALUE - Price difference between 3 and 5 door cars are not great.


### **Origin_country** (118,312 / 208,304 non-null)

**Description:** Contains information about car's origin country

**Importance:** Not very important as the most of the cars on Polish market have european origin, with similar climate and regulations.

**Handling N/A:** REMOVE COLUMN


### **First_owner** (65,094 / 208,304 non-null)

**Description:** Contains information about if car had only one owner.

**Importance:** Not very important and high percentage of missing values

**Handling N/A:** NULL = NO


### **First_registration_date** (86,445 / 208,304 non-null)

**Description:** Contains information about cars first registration date

**Importance:** Not very important and high percentage of missing values

**Handling N/A:** REMOVE COLUMN


### **Offer_location** (208,304 / 208,304 non-null)
**Description:** Contains information about offer location

**Importance:** Not important

**Handling N/A:** REMOVE COLUMN

In [104]:
# Drop rows that were designated for deletion

# Remove rows where Mileage_km is missing
df = df.dropna(subset=["Mileage_km"])

# Remove rows where Power_HP is missing
df = df.dropna(subset=["Power_HP"])

# Remove rows where Displacement_cm3 is missing
df = df.dropna(subset=["Displacement_cm3"])

In [105]:
# Handle null values in the remaining columns

df["Drive"] = df.groupby("Vehicle_model")["Drive"].transform(
    lambda x: x.fillna(x.mode()[0] if not x.mode().empty else "Front wheels")
)

df["Transmission"] = df.groupby("Vehicle_model")["Transmission"].transform(
    lambda x: x.fillna(x.mode()[0] if not x.mode().empty else "Manual")
)

df["Doors_number"] = df.groupby("Vehicle_model")["Doors_number"].transform(
    lambda x: x.fillna(x.mode()[0] if not x.mode().empty else 5)
)

df["First_owner"] = df["First_owner"].fillna(False).replace("Yes", True).infer_objects(copy=False)

df["Vehicle_generation"] = df["Vehicle_generation"].fillna("Unknown")

# Convert to string to avoid TypeError with categorical data
df["Vehicle_model"] = df["Vehicle_model"].astype(str)
df["Vehicle_version"] = df["Vehicle_version"].fillna("").astype(str)

df["Advanced_model"] = df["Vehicle_model"] + " " + df["Vehicle_version"].fillna("")
df["Advanced_model"] = df["Advanced_model"].str.strip()

df.head(50)

/var/folders/_n/377pmz811jbcznwy1kyq7svh0000gn/T/ipykernel_35315/2005879412.py:9: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['First_owner'] = df['First_owner'].fillna(False).replace('Yes', True).infer_objects(copy=False)


Index   Price Currency Condition Vehicle_brand Vehicle_model                Vehicle_version Vehicle_generation  Production_year  Mileage_km  Power_HP  Displacement_cm3 Fuel_type  CO2_emissions         Drive Transmission         Type  Doors_number  Colour Origin_country  First_owner First_registration_date Offer_publication_date                                     Offer_location                                           Features              Vehicle                     Advanced_model
0       0   86200      PLN       New        Abarth           595                                           Unknown             2021         1.0     145.0            1400.0  Gasoline            NaN  Front wheels       Manual   small_cars           3.0    gray            NaN        False                     NaN             04/05/2021  ul. Jubilerska 6 - 04-190 Warszawa, Mazowiecki...                                                 []           Abarth/595                                595
1       1   43500      PLN      Used        Abarth         Other                                           Unknown             1974     59000.0      75.0            1100.0  Gasoline            NaN  Front wheels       Manual        coupe           2.0  silver            NaN        False                     NaN             03/05/2021  kanonierska12 - 04-425 Warszawa, Rembertów (Po...                                                 []         Abarth/Other                              Other
2       2   44900      PLN      Used        Abarth           500                                           Unknown             2018     52000.0     180.0            1368.0  Gasoline            NaN  Front wheels    Automatic   small_cars           3.0  silver            NaN        False                     NaN             03/05/2021                   Warszawa, Mazowieckie, Białołęka  ['ABS', 'Electric front windows', 'Drivers air...           Abarth/500                                500
3       3   39900      PLN      Used        Abarth           500                                           Unknown             2012     29000.0     160.0            1368.0  Gasoline          139.0  Front wheels       Manual   small_cars           3.0    gray            NaN        False                     NaN             30/04/2021                                  Jaworzno, Śląskie  ['ABS', 'Electric front windows', 'Drivers air...           Abarth/500                                500
4       4   97900      PLN       New        Abarth           595                                           Unknown             2021       600.0     165.0            1368.0  Gasoline            NaN  Front wheels       Manual   small_cars           3.0    blue            NaN        False                     NaN             30/04/2021  ul. Gorzysława 9 - 61-057 Poznań, Nowe Miasto ...  ['ABS', 'Electrically adjustable mirrors', 'Pa...           Abarth/595                                595
5       5   62950      PLN      Used        Abarth           595                                           Unknown             2016     46060.0     180.0            1368.0  Gasoline          139.0  Front wheels       Manual   small_cars           3.0    gray        Germany         True              15/03/2016             30/04/2021  Modlińska 157 - 03-186 Warszawa, Białołęka (Po...  ['ABS', 'Electric front windows', 'Drivers air...           Abarth/595                                595
6       6   69900      PLN      Used        Abarth           595                                           Unknown             2021      2900.0     145.0            1368.0  Gasoline            NaN  Front wheels       Manual   small_cars           2.0   black        Germany        False              25/02/2021             29/04/2021                             Żary, żarski, Lubuskie                                                 []           Abarth/595                                595
7       7   69900      PLN      Used        Abarth           500    

In [106]:
# Remove columns that were designated for deletion
columns_to_remove = [
    "CO2_emissions",
    "Origin_country",
    "First_registration_date",
    "Offer_location",
    "Vehicle",
    "Vehicle_version",
]

df = df.drop(columns=columns_to_remove)

# Features
Extract unique values first to evaluate the data.


In [107]:
import ast

# Extract all unique feature values from the 'Features' column
unique_features = set()

# Iterate through each row in the 'Features' column
for features_str in df["Features"]:
    # Convert the string representation of the list to an actual list
    features_list = ast.literal_eval(features_str)
    unique_features.update(features_list)

# Convert the set to a sorted list
unique_features = sorted(unique_features)

# Print all unique feature values
print(unique_features)

['ABS', 'ASR (traction control)', 'AUX socket', 'Active cruise control', 'Adjustable suspension', 'Aftermarket radio', 'Air curtains', 'Airbag protecting the knees', 'Alarm', 'Alloy wheels', 'Automatic air conditioning', 'Auxiliary heating', 'Blind spot sensor', 'Bluetooth', 'CD', 'CD changer', 'Central locking', 'Cruise control', 'DVD player', 'Daytime running lights', 'Drivers airbag', 'Dual zone air conditioning', 'ESP(stabilization of the track)', 'Electric front windows', 'Electric rear windows', 'Electrically adjustable mirrors', 'Electrically adjustable seats', 'Electrochromic rear view mirror', 'Electrochromic side mirrors', 'Factory radio', 'Fog lights', 'Four-zone air conditioning', 'Front parking sensors', 'Front side airbags', 'GPS navigation', 'HUD(head-up display)', 'Heated front seats', 'Heated rear seats', 'Heated side mirrors', 'Heated windscreen', 'Hook', 'Immobilizer', 'Isofix', 'LED lights', 'Lane assistant', 'Leather upholstery', 'MP3', 'Manual air conditioning', '

As the features column contains a normalized list of features (there is not two different values for ABS), we will process it by using one-hot encoding.

In [108]:
# Convert the string representation of the list to an actual list
df["Features"] = df["Features"].apply(ast.literal_eval)

# Convert the list of features into additional boolean columns
features_dummies = df["Features"].str.join("|").str.get_dummies()

# Concatenate the original DataFrame with the new boolean columns
df = pd.concat([df, features_dummies], axis=1)

# Drop the original 'Features' column if no longer needed
df = df.drop(columns=["Features"])

# Display the result
df.head()

Index  Price Currency Condition Vehicle_brand Vehicle_model Vehicle_generation  Production_year  Mileage_km  Power_HP  Displacement_cm3 Fuel_type         Drive Transmission        Type  Doors_number  Colour  First_owner Offer_publication_date Advanced_model  ABS  ASR (traction control)  AUX socket  Active cruise control  Adjustable suspension  Aftermarket radio  Air curtains  Airbag protecting the knees  Alarm  Alloy wheels  Automatic air conditioning  Auxiliary heating  Blind spot sensor  Bluetooth  CD  CD changer  Central locking  Cruise control  DVD player  Daytime running lights  Drivers airbag  Dual zone air conditioning  ESP(stabilization of the track)  Electric front windows  Electric rear windows  Electrically adjustable mirrors  Electrically adjustable seats  Electrochromic rear view mirror  Electrochromic side mirrors  Factory radio  Fog lights  Four-zone air conditioning  Front parking sensors  Front side airbags  GPS navigation  HUD(head-up display)  \
0      0  86200      PLN       New        Abarth           595            Unknown             2021         1.0     145.0            1400.0  Gasoline  Front wheels       Manual  small_cars           3.0    gray        False             04/05/2021            595    0                       0           0                      0                      0                  0             0                            0      0             0                           0                  0                  0          0   0           0                0               0           0                       0               0                           0                                0                       0                      0                                0                              0                                0                            0              0           0                           0                      0                   0               0                     0   
1      1  43500      PLN      Used        Abarth         Other            Unknown             1974     59000.0      75.0            1100.0  Gasoline  Front wheels       Manual       coupe           2.0  silver        False             03/05/2021          Other    0                       0           0                      0                      0                  0             0                            0      0             0                           0                  0                  0          0   0           0                0               0           0                       0               0                           0                                0                       0                      0                                0                              0                                0                            0              0           0                           0                      0                   0               0                     0   
2      2  44900      PLN      Used        Abarth           500            Unknown             2018     52000.0     180.0            1368.0  Gasoline  Front wheels    Automatic  small_cars           3.0  silver        False             03/05/2021            500    1                       1           0                      0                      0                  0             0                            1      1             1                           1                  0                  0          1   1           0                1               0           0                       0               1                           0                                0                       1                      0                                1                              0                                0                            0              1           0                           0                      0                   0               0                     0   
3      3  39900      PLN      Used        Abarth           500            U

## Types
This mapping is designated for initial table. All fields generated by one-hot reloading are boolean values,

In [109]:
dtype_mapping = {
    "Index": "Int64",
    "Price": "Int64",
    "Currency": "category",
    "Condition": "category",
    "Vehicle_brand": "category",
    "Vehicle_model": "category",
    "Production_year": "Int64",
    "Mileage_km": "Int64",
    "Power_HP": "Int64",
    "Displacement_cm3": "Int64",
    "Fuel_type": "category",
    "Drive": "category",
    "Transmission": "category",
    "Type": "category",
    "Doors_number": "Int64",
    "Colour": "category",
    "First_owner": "boolean",
}

# Apply the dtype mapping to the DataFrame
df = df.astype(dtype_mapping)
df.head()

Index  Price Currency Condition Vehicle_brand Vehicle_model Vehicle_generation  Production_year  Mileage_km  Power_HP  Displacement_cm3 Fuel_type         Drive Transmission        Type  Doors_number  Colour  First_owner Offer_publication_date Advanced_model  ABS  ASR (traction control)  AUX socket  Active cruise control  Adjustable suspension  Aftermarket radio  Air curtains  Airbag protecting the knees  Alarm  Alloy wheels  Automatic air conditioning  Auxiliary heating  Blind spot sensor  Bluetooth  CD  CD changer  Central locking  Cruise control  DVD player  Daytime running lights  Drivers airbag  Dual zone air conditioning  ESP(stabilization of the track)  Electric front windows  Electric rear windows  Electrically adjustable mirrors  Electrically adjustable seats  Electrochromic rear view mirror  Electrochromic side mirrors  Factory radio  Fog lights  Four-zone air conditioning  Front parking sensors  Front side airbags  GPS navigation  HUD(head-up display)  \
0      0  86200      PLN       New        Abarth           595            Unknown             2021           1       145              1400  Gasoline  Front wheels       Manual  small_cars             3    gray        False             04/05/2021            595    0                       0           0                      0                      0                  0             0                            0      0             0                           0                  0                  0          0   0           0                0               0           0                       0               0                           0                                0                       0                      0                                0                              0                                0                            0              0           0                           0                      0                   0               0                     0   
1      1  43500      PLN      Used        Abarth         Other            Unknown             1974       59000        75              1100  Gasoline  Front wheels       Manual       coupe             2  silver        False             03/05/2021          Other    0                       0           0                      0                      0                  0             0                            0      0             0                           0                  0                  0          0   0           0                0               0           0                       0               0                           0                                0                       0                      0                                0                              0                                0                            0              0           0                           0                      0                   0               0                     0   
2      2  44900      PLN      Used        Abarth           500            Unknown             2018       52000       180              1368  Gasoline  Front wheels    Automatic  small_cars             3  silver        False             03/05/2021            500    1                       1           0                      0                      0                  0             0                            1      1             1                           1                  0                  0          1   1           0                1               0           0                       0               1                           0                                0                       1                      0                                1                              0                                0                            0              1           0                           0                      0                   0               0                     0   
3      3  39900      PLN      Used        Abarth           500            U

In [110]:
# Export the resulting DataFrame to a CSV file
output_filepath = "../scripts/data/"
output_filename = "processed_car_sale_ads.csv"
df.to_csv(output_filepath + output_filename, index=False)

# Display a message indicating the file has been saved
print(f"Resulting table has been exported to {output_filepath + output_filename}")

Resulting table has been exported to ../scripts/data/processed_car_sale_ads.csv
